File: test_login.py
"""
@fileoverview Teste de login com dados válidos
@directory dom-v1
@description Script para testar login com dados reais do banco
@created 2024-12-19
@lastModified 2024-12-19
@author Equipe DOM v1
"""

import requests
import json

def test_login():
    """Testa o login com dados válidos"""
    try:
        print("=== Teste de Login ===")
        
        # URL do backend
        url = "http://localhost:8000/api/auth/login"
        
        # Dados de teste (usando um dos usuários criados no setup)
        login_data = {
            "cpf": "303.617.927-51",  # Maria Empregadora
            "password": "123456",
            "profile": "empregador"
        }
        
        print(f"Tentando login com CPF: {login_data['cpf']}")
        
        # Fazer requisição
        response = requests.post(url, json=login_data)
        
        print(f"Status Code: {response.status_code}")
        print(f"Response: {response.text}")
        
        if response.status_code == 200:
            data = response.json()
            print("✅ Login bem-sucedido!")
            print(f"Usuário: {data.get('name')}")
            print(f"Perfil: {data.get('profile')}")
            print(f"Token: {data.get('access_token')[:20]}...")
            print(f"Foto: {'Sim' if data.get('photo') else 'Não'}")
        else:
            print("❌ Login falhou")
            
        return response.status_code == 200
        
    except Exception as e:
        print(f"❌ Erro no teste: {e}")
        return False

if __name__ == "__main__":
    success = test_login()
    exit(0 if success else 1) 

File: frontend/test_login.js
/**
 * @fileoverview Teste de Login
 * @description Script para testar o fluxo de login e identificar problemas
 */

// Simular dados de login
const testCredentials = {
  cpf: '12345678901',
  password: '123456',
  remember_me: false
}

// Testar o endpoint de login
async function testLogin() {
  console.log('🧪 Testando login...')
  
  try {
    const response = await fetch('/api/auth/login', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(testCredentials)
    })
    
    console.log('📡 Status da resposta:', response.status)
    
    if (response.ok) {
      const data = await response.json()
      console.log('✅ Login bem-sucedido:', data)
      
      // Testar se os dados estão sendo salvos corretamente
      console.log('💾 Salvando dados no localStorage...')
      localStorage.setItem('userToken', data.access_token)
      localStorage.setItem('userData', JSON.stringify(data))
      
      console.log('🔍 Verificando dados salvos:')
      console.log('Token:', localStorage.getItem('userToken'))
      console.log('User Data:', localStorage.getItem('userData'))
      
      // Testar redirecionamento
      console.log('🔄 Testando redirecionamento...')
      window.location.href = '/dashboard?profile=' + (data.profile || 'empregador')
      
    } else {
      const errorData = await response.json()
      console.log('❌ Erro no login:', errorData)
    }
  } catch (error) {
    console.log('💥 Erro na requisição:', error)
  }
}

// Executar teste
testLogin() 